File: MasterDirectory/MasterDirectory.Web/Modules/ServerTypes/UltimaMilla/CategoriaUltimaMillaForm.ts
import { StringEditor, LookupEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface CategoriaUltimaMillaForm {
    LocalSap: StringEditor;
    Estado: LookupEditor;
    Prov99Min: LookupEditor;
    ProvMu: LookupEditor;
    ProvCid: LookupEditor;
    ProvRappiCargo: LookupEditor;
    VentaTelf99Min: LookupEditor;
    VentaTelfMu: LookupEditor;
    VentaTelfCid: LookupEditor;
    VentaTelfRappiCargo: LookupEditor;
    Garantizado99Min: LookupEditor;
    GarantizadoMu: LookupEditor;
    GarantizadoCid: LookupEditor;
    ECommerceDelivery: LookupEditor;
    ECommerceClickCollect: LookupEditor;
    ECommerceTipo: LookupEditor;
    Fijo99Min: LookupEditor;
    FijoMu: LookupEditor;
    FijoCid: LookupEditor;
    OnDemandMu: LookupEditor;
    OnDemandRappiCargo: LookupEditor;
    CanalesDigitalesRappi: LookupEditor;
    CanalesDigitalesUber: LookupEditor;
    ServicioEfectivo1: LookupEditor;
    ServicioEfectivo2: LookupEditor;
    ServicioEfectivo3: LookupEditor;
    ServicioEfectivo4: LookupEditor;
    ServicioEfectivo5: LookupEditor;
    ServicioEfectivo6: LookupEditor;
    ServicioEfectivo7: LookupEditor;
    ServicioEfectivo8: LookupEditor;
    ServicioTarjeta9: LookupEditor;
    ServicioTarjeta10: LookupEditor;
    ServicioTarjeta11: LookupEditor;
    ServicioTarjeta12: LookupEditor;
    InicioServicio: StringEditor;
    CierreServicio: StringEditor;
}

export class CategoriaUltimaMillaForm extends PrefixedContext {
    static formKey = 'UltimaMilla.CategoriaUltimaMilla';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!CategoriaUltimaMillaForm.init)  {
            CategoriaUltimaMillaForm.init = true;

            var w0 = StringEditor;
            var w1 = LookupEditor;

            initFormType(CategoriaUltimaMillaForm, [
                'LocalSap', w0,
                'Estado', w1,
                'Prov99Min', w1,
                'ProvMu', w1,
                'ProvCid', w1,
                'ProvRappiCargo', w1,
                'VentaTelf99Min', w1,
                'VentaTelfMu', w1,
                'VentaTelfCid', w1,
                'VentaTelfRappiCargo', w1,
                'Garantizado99Min', w1,
                'GarantizadoMu', w1,
                'GarantizadoCid', w1,
                'ECommerceDelivery', w1,
                'ECommerceClickCollect', w1,
                'ECommerceTipo', w1,
                'Fijo99Min', w1,
                'FijoMu', w1,
                'FijoCid', w1,
                'OnDemandMu', w1,
                'OnDemandRappiCargo', w1,
                'CanalesDigitalesRappi', w1,
                'CanalesDigitalesUber', w1,
                'ServicioEfectivo1', w1,
                'ServicioEfectivo2', w1,
                'ServicioEfectivo3', w1,
                'ServicioEfectivo4', w1,
                'ServicioEfectivo5', w1,
                'ServicioEfectivo6', w1,
                'ServicioEfectivo7', w1,
                'ServicioEfectivo8', w1,
                'ServicioTarjeta9', w1,
                'ServicioTarjeta10', w1,
                'ServicioTarjeta11', w1,
                'ServicioTarjeta12', w1,
                'InicioServicio', w0,
                'CierreServicio', w0
            ]);
        }
    }
}

File: MasterDirectory/MasterDirectory.Web/Modules/UltimaMilla/CategoriaUltimaMilla/CategoriaUltimaMillaForm.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;

namespace MasterDirectory.UltimaMilla.Forms;

[FormScript("UltimaMilla.CategoriaUltimaMilla")]
[BasedOnRow(typeof(CategoriaUltimaMillaRow), CheckNames = true)]
public class CategoriaUltimaMillaForm
{
    [Tab("General")]
    public string LocalSap { get; set; }
    public string Estado { get; set; }
    [Tab("Proveedores")]

    [DisplayName("99Min")]
    public string Prov99Min { get; set; }
    [DisplayName("MU")]
    public string ProvMu { get; set; }
    [DisplayName("CID")]
    public string ProvCid { get; set; }
    [DisplayName("Rappi Cargo")]
    public string ProvRappiCargo { get; set; }
    [Tab("Venta Telefónica")]
    [DisplayName("99Min")]
    public string VentaTelf99Min { get; set; }
    [DisplayName("MU")]
    public string VentaTelfMu { get; set; }
    [DisplayName("CID")]
    public string VentaTelfCid { get; set; }
    [DisplayName("Rappi Cargo")]
    public string VentaTelfRappiCargo { get; set; }
    [Tab("Garantizado")]
    [DisplayName("99Min")]
    public string Garantizado99Min { get; set; }
    [DisplayName("MU")]
    public string GarantizadoMu { get; set; }
    [DisplayName("CID")]
    public string GarantizadoCid { get; set; }
    [Tab("E-Commerce")]
    [DisplayName("Delivery")]
    public string ECommerceDelivery { get; set; }
    [DisplayName("Click&Collect")]
    public string ECommerceClickCollect { get; set; }
    [DisplayName("Tipo")]
    public string ECommerceTipo { get; set; }
    [Tab("Fijo")]
    [DisplayName("99Min")]
    public string Fijo99Min { get; set; }
    [DisplayName("MU")]
    public string FijoMu { get; set; }
    [DisplayName("CID")]
    public string FijoCid { get; set; }
    [Tab("On Demand")]
    [DisplayName("MU")]
    public string OnDemandMu { get; set; }
    [DisplayName("Rappi Cargo")]
    public string OnDemandRappiCargo { get; set; }
    [Tab("Canales Digitales")]
    [DisplayName("Rappi")]
    public string CanalesDigitalesRappi { get; set; }
    [DisplayName("Uber")]
    public string CanalesDigitalesUber { get; set; }
    [Tab("Tipo de Servicio - Efectivo")]
    [DisplayName("1")]
    public string ServicioEfectivo1 { get; set; }
    [DisplayName("2")]
    public string ServicioEfectivo2 { get; set; }
    [DisplayName("3")]
    public string ServicioEfectivo3 { get; set; }
    [DisplayName("4")]
    public string ServicioEfectivo4 { get; set; }
    [DisplayName("5")]
    public string ServicioEfectivo5 { get; set; }
    [DisplayName("6")]
    public string ServicioEfectivo6 { get; set; }
    [DisplayName("7")]
    public string ServicioEfectivo7 { get; set; }
    [DisplayName("8")]
    public string ServicioEfectivo8 { get; set; }
    [Tab("Tipo de Servicio - Tarjeta")]
    [DisplayName("9")]
    public string ServicioTarjeta9 { get; set; }
    [DisplayName("10")]
    public string ServicioTarjeta10 { get; set; }
    [DisplayName("11")]
    public string ServicioTarjeta11 { get; set; }
    [DisplayName("12")]
    public string ServicioTarjeta12 { get; set; }
    [Tab("Horario")]
    public string InicioServicio { get; set; }
    public string CierreServicio { get; set; }
    //public DateTime DtRegistro { get; set; }
}